Module01/Lexer.py

```diff
@@ -1,288 +1 @@
-"""
-Project01 - CS5400
-Group Members: Daniel Hayes and Sarah Crane
-Semptember 18, 2024
-"""
-class Lexer:
-    """
-    Class Lexer
-    Processes input strings and produces tokens
-    """
-    def __init__(self, text):
-        """
-        Initializes the lexer with the input text.
-        Sets the initial position and current character.
-        """
-        self.text = text
-        self.charlist = list(text)  # Creates list of chars out of String input
-        self.current_char = self.charlist[0]    # Stores current char
-        self.position = 0       # Stores int position in char list
-        self.current_token = "" # Stores String of current token
-        self.period_counter = 0 # Counts periods in numbers (cannot have more than 1)
-        self.is_alpha = False   # Currently reading variable boolean - is True if going through variable
-        self.is_number = False  # Currently reading numerical boolean - is True is going through number
-        self.last_token_type = None # Stores previous token type for potential error handling
-        self.past_assign_op = False # Stores whether or not past = sign
 
-    def error(self,a):
-        """
-        Raises an exception for invalid characters encountered
-        during tokenization.
-        """
-        if(a == 0):
-            raise Exception("Invalid characters present.")
-        if(a == 1):
-            raise Exception("There must be a number before a '.'.")
-        if(a == 2):
-            raise Exception("There can only be one '.' in a number.")
-        if(a == 3):
-            raise Exception("Cannot have a number value prior to assignment operator.")
-        if(a == 4):
-            raise Exception("Cannot have an arithmetic expression prior to assignment operator.")
-
-    def advance(self):
-        """
-        Advances to the next character in the input text.
-        Updates the current character to the new position.
-        """
-        self.position = self.position+1
-        if self.position >= len(self.charlist):
-            self.current_char = None
-        else: 
-            self.current_char = self.charlist[self.position]
-
-    def get_next_token(self):
```

Comment:
Relying on boolean flags should be bad for future development. The number of boolean flags increases a faster than linear rate as the complexity of your programming language icnreases